### PR TITLE
- Updated Java sdk version from 1.8 to 2.0.0-beta2

### PR DESCRIPTION
**Pull request by mnoman09**
- Removed
   getVariableString
   getVariableBoolean
   getVariableInteger
   getVariableDouble
-Remove track with revenue as a parameter. Pass the revenue value as an event tag instead
   track(String, String, long)
   track(String, String, Map<String, String>, long)

### CODE REVIEW

File: android-sdk/src/main/java/com/optimizely/ab/android/sdk/OptimizelyClient.java [hunk 1]
```diff
@@ -220,7 +221,7 @@ public void track(@NonNull String eventName,
                       @NonNull String userId,
                       long eventValue) throws UnknownEventTypeException {
         if (isValid()) {
-            optimizely.track(eventName, userId, getDefaultAttributes(), eventValue);
+            optimizely.track(eventName, userId, getDefaultAttributes(), Collections.singletonMap(ReservedEventKey.REVENUE.toString(), eventValue));
```
Comment (raheeladnan) [inline android-sdk/src/main/java/com/optimizely/ab/android/sdk/OptimizelyClient.java]: Good one (Y)

File: datafile-handler/build.gradle [hunk 2]
```diff
@@ -32,6 +32,7 @@ android {
         versionName version_name
         testInstrumentationRunner "android.support.test.runner.AndroidJUnitRunner"
         consumerProguardFiles '../proguard-rules.txt'
+        multiDexEnabled true
```
Comment (raheeladnan) [inline datafile-handler/build.gradle]: Everything seems fine, you just need to configure multidex properly. Our  minimum SDK version is less than 20 so we need to use multidex support library. Change Application of demo app as well and inherit that from MultidexApplication.

File: android-sdk/build.gradle [hunk 3]
```diff
@@ -66,6 +66,8 @@ dependencies {
     compile "com.optimizely.ab:core-api:$java_core_ver"
     provided "com.android.support:support-annotations:$support_annotations_ver"
 
+    compile 'com.android.support:multidex:1.0.1'
```
Comment (raheeladnan) [inline android-sdk/build.gradle]: Don't use versions inline specially when you need to define same dependency in multiple gradle files. Follow [this link](https://segunfamisa.com/posts/android-gradle-extra-properties), you can also see this is already implemented. For example _compile "com.optimizely.ab:core-api:$java_core_ver"_ this java_core_version is defined in Project level buile.gradle file. 
You need to define _multidex_ver = "1.0.1"_ in "ext" in Project level build.gradle file and use _compile "com.android.support:multidex:$multidex_ver"_ in all other files.

Comment (mnoman09) [reply android-sdk/build.gradle]: Done

**Pull request merged**